features/topology-map/plugins/org.opennms.features.topology.plugins.topo.application/src/main/java/org/opennms/features/topology/plugins/topo/application/GuiApplicationVertex.java

```diff
@@ -30,39 +30,56 @@
 
 import org.opennms.features.topology.api.topo.AbstractLevelAwareVertex;
 import org.opennms.features.topology.api.topo.LevelAware;
+import org.opennms.netmgt.graph.provider.application.ApplicationVertex;
 import org.opennms.netmgt.model.OnmsApplication;
 import org.opennms.netmgt.model.OnmsMonitoredService;
 import org.opennms.netmgt.model.OnmsServiceType;
 
-public class ApplicationVertex extends AbstractLevelAwareVertex implements LevelAware {
+public class GuiApplicationVertex extends AbstractLevelAwareVertex implements LevelAware {
```

Comment:
@mvr we should discuss naming here. This class was previously called ApplicationVertex. But we have now a org.opennms.netmgt.graph.provider.application.ApplicationVertex. To avoid confusion I renamed it to GuiApplicationVertex. Not sure if that is a good name.
Maybe this class should be names BackendApplicationVertex since it belongs to the BackendGraph (although I don't really like this name since it holds Frontend specific attributes such as Tooltips).

Reply:
This was meant for @mvrueden 

(I think you've just introduced me to who my competition is for the mvr username on other websites!)

Reply:
@mvr sorry!

Reply:
@mvr everyone needs a villain :)

Reply:
Yeah, the naming is bad.
The problem here is, that we have various `Graph` classes:
 - `org.opennms.features.topology.api.Graph`. This is actually the UI representation of the Graph
- `org.opennms.features.topology.api.topo.Graph`. The graph provided by the `TopologyProvider`. To not have two classes named `Graph`, I decided to name this `BackendGraph`. Probably better to name the UI part, just `UIGraph` or `GraphSnippet`.
 - `org.opennms.netmgt.graph.api.Graph`. The new `Graph` interface actually representing a Graph.

The naming `GuiApplicationVertex` is misleading and not correct. 
It is an `ApplicationVertex` within the `ApplicationTopologyProvider`. 
We already have an `ApplicationVertex` in `ApplicationGraphProvider`, so I decided to for now name it `OnmsApplicationVertex`. Alternatively we could bane it `TopologyApplicationVertex` or something else.
We should come up with a consistent naming scheme, as we will have the same problem with all the other providers as well. We could also just name them the same and distinguish only via full name `a.A` vs `b.A`

Reply:
All the classes in `features/topology` are supposed to be "the old way" and all classes in `features/graph` are supposed to be "the new way". How about just prefix everything from "the old way" with `Legacy`. Then we don't have the naming collisions anymore.